File: app/src/main/java/com/essence/mvvmapplication/posts/domain/PostsRepository.java
package com.essence.mvvmapplication.posts.domain;

import com.essence.mvvmapplication.posts.model.PostModel;

import java.util.List;

import javax.inject.Inject;
import javax.inject.Singleton;

import io.reactivex.rxjava3.core.Observable;

@Singleton
public class PostsRepository {
    private static final String TAG = PostsRepository.class.getSimpleName();

    @Inject
    PostsServices postsServices;

    @Inject
    public PostsRepository() {
    }

    public Observable<List<PostModel>> getPosts() {
        return postsServices.getPosts();
    }
}

File: app/src/main/java/com/essence/mvvmapplication/posts/domain/PostsRepository.kt
package com.essence.mvvmapplication.posts.domain

import com.essence.mvvmapplication.posts.model.PostModel
import io.reactivex.rxjava3.core.Observable
import javax.inject.Inject
import javax.inject.Singleton

@Singleton
class PostsRepository @Inject constructor() {
    @JvmField
    @Inject
    var postsServices: PostsServices? = null
    val posts: Observable<List<PostModel?>?>?
        get() = postsServices!!.posts

    companion object {
        private val TAG = PostsRepository::class.java.simpleName
    }
}